source/postprocess/velocity_boundary_statistics.cc

```diff
@@ -76,48 +80,78 @@ namespace aspect
                   {
                     const double vel_mag = velocities[q].norm();
                     local_max = std::max(vel_mag,
                                          local_max);
                     local_min = std::min(vel_mag,
                                          local_min);
+                    local_sqvel += ((vel_mag * vel_mag) * fe_face_values.JxW(q));
+                    local_fe_face_area += fe_face_values.JxW(q);
                   }
-                // then merge them with the min/max velocities we found for other faces with the same boundary indicator
+
+                // then merge them with the min/max/squared velocities
+                // and face areas we found for other faces with the same boundary indicator
                 const types::boundary_id boundary_indicator
                   = cell->face(f)->boundary_id();
 
                 local_max_vel[boundary_indicator] = std::max(local_max,
                                                              local_max_vel[boundary_indicator]);
                 local_min_vel[boundary_indicator] = std::min(local_min,
                                                              local_min_vel[boundary_indicator]);
+                local_boundary_sqvel[boundary_indicator] += local_sqvel;
+                local_boundary_area[boundary_indicator] += local_fe_face_area;
               }
 
       // now communicate to get the global values
       std::map<types::boundary_id, double> global_max_vel;
       std::map<types::boundary_id, double> global_min_vel;
+      std::map<types::boundary_id, double> boundary_vel_rms;
       {
         // first collect local values in the same order in which they are listed
         // in the set of boundary indicators
         std::vector<double> local_max_values;
         std::vector<double> local_min_values;
+        std::vector<double> boundary_sqvel_values;
+        std::vector<double> boundary_area_values;
+
         for (std::set<types::boundary_id>::const_iterator
              p = boundary_indicators.begin();
              p != boundary_indicators.end(); ++p)
           {
             local_max_values.push_back (local_max_vel[*p]);
             local_min_values.push_back (local_min_vel[*p]);
+
+            boundary_sqvel_values.push_back (local_boundary_sqvel[*p]);
+            boundary_area_values.push_back (local_boundary_area[*p]);
           }
         // then collect contributions from all processors
         std::vector<double> global_max_values (local_max_values.size());
         Utilities::MPI::max (local_max_values, this->get_mpi_communicator(), global_max_values);
         std::vector<double> global_min_values (local_min_values.size());
         Utilities::MPI::min (local_min_values, this->get_mpi_communicator(), global_min_values);
 
+        std::vector<double> boundary_vel_rms_values (boundary_sqvel_values.size());
+        std::vector<double> boundary_area (boundary_area_values.size());
```

Comment:
then you can call this `global_rms_velocity_values` and `global_boundary_area_values`

Reply:
Changed boundary_vel_rms_values to global_velocity_square_integral_values. I removed rms from the variable name because the rms values are now calculated in the step that also put values back into maps (see above comment). The other variable is changed as suggested.